CONTRIBUTING.md

```diff
@@ -21,6 +23,40 @@ sending a pull request. Please follow these guidelines:
     document the setting. The comments in this file are extracted and turned into documentation. Pay
     attention to the comment formatting and follow the existing style closely.
 
+## Development Environment
+
+### Prerequisites 
+
+On Windows, make sure to set your PowerShell execution policy as shown [here](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.security/set-executionpolicy?view=powershell-5.1),
+otherwise the PowerShell tests will fail.
+
+### Setting Up
+
+To begin development on Rez you'll first need to set up your development environment. There are many 
+ways you can do it, but this is the recommended approach. This approach will automatically create a virtual
+environment for you, patch the Rez binaries, and copy completion scripts. All tests will be run this way.
+
+1. Fork the repo and clone it.
+2. Create a new Git branch and check it out.
+3. Install your local Rez code by running `python install.py venv`.
+4. Activate the virtual environment by running the `activate` file.
+5. Add the `Scripts/rez` folder on Windows or the `bin/rez` folder on Mac/Linux to the `PATH` environment variable.
+
+There is an alternative method of setting up your development environment, that doesn't use the `install.py`
+script. This method is not recommended however, as not all tests will be run this way.
```

Comment:
I wouldn't say that it's not recommended. It's more that it won't run all tests. So it's useful to test quick things, etc. We can also easily use an editable install `pip isntall -e .` which speeds up the development a little bit.